workflows: build gitian on tag

selsta opened the pull request:
- only runs on tag
- each OS uses a separate runner
- hashes get printed on summary page
- the resulting build archives get uploaded as artifacts

hyc commented on .github/workflows/gitian.yml:
```diff
@@ -0,0 +1,46 @@
+      run: |
+        curl -O https://raw.githubusercontent.com/monero-project/monero/${{ github.ref_name }}/contrib/gitian/gitian-build.py
+        chmod +x gitian-build.py
+        sudo ./gitian-build.py --setup --docker github-actions ${{ github.ref_name }}
```
Why does this need a sudo if you've added $USER to the docker group?

selsta replied:
Not sure, I got a permission error without it.

selsta replied:
Updated and removed sudo.

Malinero commented on .github/workflows/gitian.yml:
```diff
@@ -0,0 +1,46 @@
+    steps:
+    - name: prepare gitian build host 
+      run: |
+        sudo usermod -aG docker $USER
```
this seems unnecessary as $USER is already a member of group docker

selsta replied:
Removed.

SChernykh commented on .github/workflows/gitian.yml:
```diff
@@ -0,0 +1,44 @@
+
+jobs:
+  build-gitian:
+    runs-on: ubuntu-20.04
```
https://github.com/monero-project/monero/blob/master/contrib/gitian/README.md says
> Gitian host OS should be Ubuntu 18.04 "Bionic Beaver". If you are on a mac or windows for example, you can run it in a VM but will be slower.

I know that Github is going to deprecate 18.04 soon, but changing OS to 20.04 will probably create different binaries.

selsta replied:
I did confirm that the hashes match.

SChernykh replied:
The we can probably change the README as well

hyc replied:
Gitian creates its own container, built on Ubuntu 18.04 regardless. Thus the binaries will always be the same. The difference in host OS can affect the format of the .assert files, which are generated by host tools, but that won't matter if nobody is expecting to sign these builds.

selsta commented:
Updated to add `sudo apt update` to avoid a potential file not found error message during `apt install` later during the gitian process. Also split the code up in more steps.
